src/app/ui/NewTaskForm.tsx

```diff
@@ -75,6 +74,8 @@ export const NewTaskForm = ({
   const statusValue = _statusValue as WorkflowStateResponse //typecasting
   const assigneeValue = _assigneeValue as IAssigneeCombined //typecasting
   const templateValue = _templateValue as ITemplate //typecasting
+  // use temp state pattern so that we don't fall into an infinite loop of assigneeValue set -> trigger -> set
+  const [tempAssignee, setTempAssignee] = useState<IAssigneeCombined | null>(assigneeValue)
```

Comment:
When is this variable set?

Reply:
This state is initially set based on assigneeValue. assigneeValue is managed in our Redux state.

assigneeValue is set in one of these ways
- Value of Filter by assignee
- When user clicks "+" icon in workflow state
- When user selects an assignee from this selector

Reply:
In this case the infinite rerender was caused due to this flow
1. assigneeValue is set by Filter by assignee
2. Dropdown truncates assigneeValue
3. assigneeValue is inconsistent between the redux state and selector
4. We hit a network call and update assigneeValue
5. assigneeValue is truncated and we get inconsistent data. Back to step 2.

Reply:
The UI code is very tightly coupled. Issues like this will keep coming up until we fix the root cause.